department-of-veterans-affairs/abd-vro
# Mcp 1593 - fix service-data-access test

Refactor service-data-access tests

## Description

Created integration test, and moved some non unit tests in unit test directory to integration tests;
Upgrade Gradle version used in Gradle wrap from 7.2 to 7.4 in order to support Java 17. (7.2 does not support Java 17);
Added unit tests.

### What was the problem?

The tests in service-data-access cannot be run. Some tests are not unit tests and cannot be run during build process.

### How does this fix it?

Upgrade Gradle version to support Java 17. Move non unit tests to integration test.

### Jira Tickets

MCP-1593

- [MCP-000](https://amida.atlassian.net/browse/MCP-000)

## How to test this PR

- Step 1. Build the application. Go to service-data-access Jacoco directory or test directory to see the test results.
- Step 2.  ./gradlew clean integrationTest.  Go to service-data-access Jacoco directory or test directory to see the test results.

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
